File: 3895-count-digit-appearances/3895-count-digit-appearances.cs
public class Solution {
    public int CountDigitOccurrences(int[] nums, int digit) {
        string num=string.Concat(nums);
        int result=0;
        for(int i=0;i<num.Length;i++){
            if(num[i]-'0'==digit) result++;
        }
        return result;
    }
}

File: 3895-count-digit-appearances/3895-count-digit-appearances.ts
function countDigitOccurrences(nums: number[], digit: number): number {
    let num=nums.join(",");
    let result=0;
    for(let i=0;i<num.length;i++){
        if(num[i]==digit.toString()){
            result++;
        }
    }
    return result;
};